ocean_lib/ocean/ocean_compute.py

```diff
@@ -95,11 +93,12 @@ def status(self, did: str, job_id: str, wallet: Wallet) -> Dict[str, Any]:
         :param wallet: Wallet instance
         :return: dict the status for an existing compute job, keys are (ok, status, statusText)
         """
-        _, service_endpoint = self._get_service_endpoint(did)
+        asset = resolve_asset(did, metadata_cache_uri=self._config.metadata_cache_uri)
```

Comment:
I resent the extra resolves but I think these were probably introduced by @DMats. David, is there any way to pass the asset and/or service here to avoid extra resolving?

Reply:
@DMats @mariacarmina, reminding you. I added this comment 5 days ago. Did you discuss it? Did you address it?

Reply:
Do you want to replace `did` with the whole asset object?

Reply:
I can do that and modify everywhere if necessary.

Reply:
I think that makes sense. @DMats can you also give us an opinion? I'd rather use the asset object.

Reply:
I'm okay with changing the OceanCompute API to take `Asset` instead of `did`.  But I'd want to see this change applied to all of the OceanCompute methods (start, status, stop, and result)

Please note, I refactored the `status`, `result`, and `result_file` methods in PR #695